### ISSUE TITLE
S1172 argument used in local method + delegate

### ISSUE THREAD

**Issue by Bouke** [SonarSource/sonar-dotnet#4942]
### Description

S1172 is raised for arguments used by local methods. While #4770 fixed quite some instances of this, the case below is not covered.

### Repro steps

```c#
private string Foo(string bar)
{
    string F() => bar;
    Func<string> f = F;
    return f();
}
```

### Expected behavior

Should not raise warning.

### Actual behavior

S1172 is raised.

### Known workarounds

N/A

### Related information

* C#/VB.NET Plugins version: 8.29.0.36737
* Visual Studio version: Rider 2021.2.1
* MSBuild / dotnet version: Microsoft (R) Build Engine version 16.8.0 for Mono
* SonarScanner for .NET version (if used)
* Operating System: macOS / Windows

**Comment by VahidN** [comment]
It's happening again in the lasted version of `SonarAnalyzer.CSharp` Version="8.36.1.44192". All of my used arguments are marked as `S1172`, in the whole project.

**Comment by pavel-mikula-sonarsource** [comment]
@VahidN Can you provide a small reproducer?

**Comment by nrooster** [comment]
This is happening in private methods for me since a week. If i change the method to  public (for testing purpose) the warning is gone. Description is "Remove this parameter '...', whose value is ignored in the mehtod".

**Comment by amoroso81** [comment]
Same for me, it's happening since a couple of weeks.
